### PR TITLE
MT-22 Lambdas refactoring, seed.cpp

### PR DESCRIPTION
**Pull request by DariuszRosolek**

### CODE REVIEW

**Review by bartalke — APPROVED** [review]

File: src/api/crypto/Seed.cpp [hunk 1]
```diff
@@ -770,13 +755,11 @@ auto Seed::SetDefault(const Identifier& id) const noexcept -> bool
...
-    const auto exists = [&] {
-        for (const auto& [value, alias] : api_.Storage().SeedList()) {
-            if (value == seedID) { return true; }
-        }
 
-        return false;
-    }();
+    auto exists = false;
+    for (const auto& [value, alias] : api_.Storage().SeedList()) {
+        if (value == seedID) { exists = true; }
```
Comment (wiesiekpap) [inline src/api/crypto/Seed.cpp]: The logic has changed. In the original, the lambda returned with true on the first match, which was correct as we were testing the existence. Here, the value of exists will depend on the last comparison. Suggestuib: { exists = true; break; }

**Review by wiesiekpap — CHANGES_REQUESTED** [review]

File: src/api/crypto/Seed.cpp (hunk 1 above)
Comment (DariuszRosolek) [reply src/api/crypto/Seed.cpp]: The logic is the same. You described situation  exist = (value == seedID) ? true : false; In our code false is assigned only once, before the loop and condition check. Anyway break is good idea for optimization rezones and I'll add it

Comment (wiesiekpap) [reply src/api/crypto/Seed.cpp]: True.

**Review by wiesiekpap — APPROVED** [review]

File: src/api/crypto/Seed.cpp (hunk 1 above)
Comment (DariuszRosolek) [reply src/api/crypto/Seed.cpp]: done

**Pull request merged**